### DISCUSSION TITLE
Ignoring check

### DISCUSSION THREAD

**Discussion by todo-username** [squizlabs/PHP_CodeSniffer#3902] [Ideas]
How can I add an exception to a file for substring checking or exclude use or namespace checking?

For example, is "use Service/Folder1/Folder2/..." more than 120 characters?

**Comment by jrfnl** [comment]
@todo-username You can use `<exclude-pattern>file/to/file\.php$</exclude-pattern>` for a sniff in your ruleset or you can use an inline ignore annotation like `// phpcs:ignore Stnd.Cat.Sniff.Code`.

Without more detailed information about what you are trying to ignore, I can't give more specific examples.

Also see:
* https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file
* https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset

**Comment by todo-username** [comment]
@jrfnl 

I need to ignore lines starting with "use FolderName/..." and "namespace FolerName/..." as they may exceed the 120 character limit and cannot be split across multiple lines. It is also impossible to reduce the level of nesting of directories (a project feature).

I know that I can ignore lines using comment blocks, but I need to keep all these restrictions inside the config, since the config is easier to manage and can only be edited after approval.

Adding an entire file or directory is also not the best solution, since the only problems are with the 'use' and 'namespace' blocks.

**Comment by todo-username** [comment]
If I could ignore the substring search line of code - that would solve my problem. 
Being able to ignore the "usage" and "namespace" blocks would solve my problem. 
Being able to specify a specific unique error in the configuration would solve my problem.

**Comment by todo-username** [comment]
It would be possible to put this in the config as in phpstan - this would solve the problem

![image](https://github.com/squizlabs/PHP_CodeSniffer/assets/43653056/ab2859ff-1f85-45b9-aed0-11abd9139f4f)

**Comment by jrfnl** [comment]
@todo-username That does not sound like a Q&A question, but a complaint/feature request.

Having said that, I'm not sure the sniff should be adjusted.

It's a users choice to use the sniff and enforce the line length limit in the first place. Secondly, the desired [max line length is configurable](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#genericfileslinelength).

**Reply by todo-username** [reply]
@jrfnl
Okay, if it's not there now, I'll change the category

**Reply by todo-username** [reply]
Increasing the line length will not work since the length of the 'use' and 'namespace' blocks in my case can fluctuate and then checking the line length will not make sense.

**Reply by jrfnl** [reply]
Changing the category is not going to help. Discussions is not the place for feature requests.

**Reply by todo-username** [reply]
I can't create an issue for a feature.
![image](https://github.com/squizlabs/PHP_CodeSniffer/assets/43653056/1368e849-f056-4dcf-8b4b-5073b5d96ecc)
This is not a bug or security vulnerability.

**Reply by jrfnl** [reply]
![image](https://github.com/squizlabs/PHP_CodeSniffer/assets/663378/9bbb7039-984d-41cb-b8a0-539a298f4156)
